components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,348 @@
+                        } catch (IdentityOAuth2Exception e) {
+                            log.error("Error while retrieving scoped access tokens for the application " + appDTO
+                                    .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                    + " : " + e);
+                        }
+                        if (scopedToken != null) {
+                            //Revoking token from database
+                            try {
+                                OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                                        .revokeAccessTokens(new String[] { scopedToken.getAccessToken() });
+
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while revoking access tokens related for the application " + appDTO
+                                        .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                        + " : " + e);
+                            }
+                            //Revoking the oauth consent from database.
+                            try {
+                                OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                                        .revokeOAuthConsentByApplicationAndUser(
+                                                authzUser.getAuthenticatedSubjectIdentifier(), tenantDomain, username);
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while revoking access tokens related for the application " + appDTO
+                                        .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                        + " : " + e);
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Method to retrieve applications authorized for user
+     * @param authenticatedUser authenticated user info
+     * @return array of authorized applications
+     * @throws IdentityOAuthAdminException exception
+     */
+    private OAuthConsumerAppDTO[] getAppsAuthorizedByUser(AuthenticatedUser authenticatedUser)
+            throws IdentityOAuthAdminException {
+
+        OAuthAppDAO appDAO = new OAuthAppDAO();
+
+        String tenantAwareusername = authenticatedUser.getUserName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        String username = UserCoreUtil.addTenantDomainToEntry(tenantAwareusername, tenantDomain);
+
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+
+        Set<String> clientIds;
+        try {
+            clientIds = OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                    .getAllTimeAuthorizedClientIds(authenticatedUser);
```

Comment:
Don't we need to check OAuthTokenPersistenceFactory.getInstance() != null and OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO() != null here?

Reply:
It's getting handled through catch block